src/coreclr/nativeaot/System.Private.CoreLib/src/System/Runtime/InteropServices/MemoryMarshal.NativeAot.cs

```diff
@@ -20,9 +20,8 @@ public static unsafe partial class MemoryMarshal
         /// </remarks>
         [Intrinsic]
         [NonVersionable]
-        [MethodImpl(MethodImplOptions.AggressiveInlining)]
         public static ref T GetArrayDataReference<T>(T[] array) =>
```

Comment:
Does it make sense to mark `ref byte GetArrayDataReference(Array array)` as an intrinsic as well?

Reply:
For that we'd need to detect the array rank in order to get the offset to the first element and I wasn't sure how to check that.